src/mission_ui.cpp

```diff
@@ -17,239 +17,277 @@
+        adjust_selected = true;
+        ImGui::SetKeyboardFocusHere( -1 );
+        selected_mission--;
+    } else if( last_action == "DOWN" ) {
+        adjust_selected = true;
+        ImGui::SetKeyboardFocusHere( 1 );
+        selected_mission++;
+    } else if( last_action == "NEXT_TAB" || last_action == "RIGHT" ) {
+        adjust_selected = true;
+        selected_mission = 0;
+        switch_tab = selected_tab;
+        ++switch_tab;
+    } else if( last_action == "PREV_TAB" || last_action == "LEFT" ) {
+        adjust_selected = true;
+        selected_mission = 0;
+        switch_tab = selected_tab;
+        --switch_tab;
+    }
 
-        const std::vector<std::pair<tab_mode, std::string>> tabs = {
-            { tab_mode::TAB_ACTIVE, _( "ACTIVE MISSIONS" ) },
-            { tab_mode::TAB_COMPLETED, _( "COMPLETED MISSIONS" ) },
-            { tab_mode::TAB_FAILED, _( "FAILED MISSIONS" ) },
-        };
-        tabs_coords = draw_tabs( w_missions, tabs, tab );
-        draw_border_below_tabs( w_missions );
-        int x1 = 2;
-        int x2 = 2;
-        for( const std::pair<tab_mode, std::string> &t : tabs ) {
-            x2 = x1 + utf8_width( t.second ) + 1;
-            if( t.first == tab ) {
-                break;
-            }
-            x1 = x2 + 2;
+    ImGuiTabItemFlags_ flags = ImGuiTabItemFlags_None;
+
+    if( ImGui::BeginTabBar( "##TAB_BAR" ) ) {
+        flags = ImGuiTabItemFlags_None;
+        if( switch_tab == mission_ui_tab_enum::ACTIVE ) {
+            flags = ImGuiTabItemFlags_SetSelected;
+            switch_tab = mission_ui_tab_enum::num_tabs;
+        }
+        if( ImGui::BeginTabItem( _( "ACTIVE" ), nullptr, flags ) ) {
+            selected_tab = mission_ui_tab_enum::ACTIVE;
+            umissions = get_avatar().get_active_missions();
+            ImGui::EndTabItem();
+        }
+        flags = ImGuiTabItemFlags_None;
+        if( switch_tab == mission_ui_tab_enum::COMPLETED ) {
+            flags = ImGuiTabItemFlags_SetSelected;
+            switch_tab = mission_ui_tab_enum::num_tabs;
         }
-        mvwputch( w_missions, point( 40, 2 ), BORDER_COLOR, x1 < 40 && 40 < x2 ? ' ' : LINE_OXXX ); // ^|^
-        mvwputch( w_missions, point( 40, TERMY - 1 ), BORDER_COLOR, LINE_XXOX ); // _|_
+        if( ImGui::BeginTabItem( _( "COMPLETED" ), nullptr, flags ) ) {
+            selected_tab = mission_ui_tab_enum::COMPLETED;
+            umissions = get_avatar().get_completed_missions();
+            ImGui::EndTabItem();
+        }
+        flags = ImGuiTabItemFlags_None;
+        if( switch_tab == mission_ui_tab_enum::FAILED ) {
+            flags = ImGuiTabItemFlags_SetSelected;
+            switch_tab = mission_ui_tab_enum::num_tabs;
+        }
+        if( ImGui::BeginTabItem( _( "FAILED" ), nullptr, flags ) ) {
+            selected_tab = mission_ui_tab_enum::FAILED;
+            umissions = get_avatar().get_failed_missions();
+            ImGui::EndTabItem();
+        }
+        ImGui::Separator();
```

Comment:
Does it need a separator here? I think the tab bar draws one as well…

Reply:
With:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/84619419/3b08efe7-cbe1-43b0-8851-43d6bcf73287)

Without:
![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/84619419/76d9bcfd-5da3-4cda-957a-cbdb81d880d8)

Hm, sure. Removed in the next push